### ISSUE TITLE
Feat: SDM iterative convergence retrieval — multi-pass search that cleans up degraded cues, grounded in Kanerva + NeurIPS 2022 attention equivalence

### ISSUE THREAD

**Issue by TSchonleber** [TSchonleber/brainctl#36]
## The finding

Bricken & Pehlevan (arXiv 2111.05498, NeurIPS 2021) proved that Transformer attention approximates Sparse Distributed Memory (Kanerva 1988) when key-query dimensionality and softmax temperature satisfy specific conditions. They verified this empirically on pre-trained GPT-2 attention heads — the attention mechanism *is* doing SDM-style content-addressable retrieval.

This means brainctl's current vector search is already performing approximate SDM. What SDM adds beyond standard nearest-neighbor search is **iterative convergence**: the output of one read can become the input of the next, progressively cleaning up degraded cues until the system converges on the stored pattern closest to the initial query.

## Why iterative convergence matters

Standard nearest-neighbor retrieval assumes the query is clean. The query is mapped once to an embedding, nearest neighbors are returned, and retrieval is done.

For agent memory, queries are often degraded:
- "What was that thing we decided about the API endpoint... something about caching?" (partial cue, no precise content)
- Retrieval starting from a noisy cue returns a marginally relevant result — but that result, used as the next query, can converge toward the correct stored memory

SDM handles this because it is an *attractor* network: the space of stored patterns acts as attractors in the high-dimensional embedding space. Iterative SDM retrieval follows a gradient descent toward the nearest attractor from wherever the initial cue lands.

## Implementation

Add optional `iterative_retrieval` mode to `tool_vsearch`:

```python
def vsearch_iterative(
    query_embedding: bytes,
    max_iter: int = 3,
    convergence_threshold: float = 0.01,
    k: int = 10,
) -> list[dict]:
    current_query = query_embedding
    prev_centroid = None

    for _ in range(max_iter):
        # Standard vec search
        results = vec_search(current_query, k=k)

        # Compute weighted centroid of top results
        scores = [r['score'] for r in results]
        embeddings = [fetch_embedding(r['id']) for r in results]
        centroid = weighted_average(embeddings, weights=scores)

        # Check convergence
        if prev_centroid and cosine_distance(centroid, prev_centroid) < convergence_threshold:
            break

        current_query = centroid
        prev_centroid = centroid

    return vec_search(current_query, k=k)
```

This runs at most 3 vec searches (each O(log N) with the existing index). On degraded queries it converges in 2-3 iterations. On clean queries it converges in 1 iteration (centroid ≈ initial query).

## What changes in retrieval results

**Clean precise query**: identical results to current behavior (convergence after 1 pass)

**Degraded/partial query**: initial pass returns marginal results; centroid of those results is closer to the "true" memory cluster; second pass returns significantly better results. This is the attractor basin effect — even a query that lands outside the basin of the target memory can converge into it if the basin is large enough.

## MCP interface

```
vsearch(query="...", iterative=true, max_iter=3, convergence_threshold=0.01)
```

Results include `converged_in_n_iterations` field so callers know whether convergence was clean.

## Schema dependencies

Iterative retrieval requires storing raw embeddings per memory (not just the vec0 virtual table index). Add `embeddings(source_table, source_id, vector BLOB)` — this table already exists in the schema.

## References
- Kanerva (1988) *Sparse Distributed Memory* — SDM foundational text
- Bricken & Pehlevan (2021) arXiv 2111.05498, NeurIPS — Transformer attention ≈ SDM (verified on GPT-2)
- Hopfield (1982) — Attractor networks and content-addressable memory (related attractor mechanism)

**Comment by TSchonleber** [comment]
Narrowing this issue. The full Kanerva SDM (sparse distributed memory with binary address space and Hamming distances) is not implementable with the current SQLite+FTS5 stack. The mathematically equivalent operation that IS implementable: iterative FTS refinement — using first-pass retrieval results to construct a richer query for a second pass.

Narrowing to: add a `multi_pass` boolean to `memory_search`. When enabled: (1) run the initial FTS query, (2) extract key terms from the top-3 results, (3) combine with original query for a second-pass FTS hit, (4) return merged + deduplicated results ranked by appearance across both passes. This is the SDM convergence behavior (noisy cue cleans up toward attractor) implemented with FTS. No vector ops required.

**Comment by TSchonleber** [comment]
Implemented via PR #48. multi_pass=true added to memory_search — SDM-style iterative FTS convergence.
